libs/frontend/util-edit-text/src/lib/prosemirror-setup.ts

```diff
@@ -12,14 +12,27 @@ export interface IProsemirrorSetupOptions {
   schema: Schema
   mapKeys?: { [key: string]: Command }
   plugins?: EditorStateConfig['plugins']
+  overwriteBaseKeys?: string[]
 }
 
 export function prosemirrorSetup(options: IProsemirrorSetupOptions): EditorState {
-  const { content: contentOption, mapKeys, plugins: pluginsOption, schema } = options
+  const {
+    content: contentOption,
+    schema,
+    mapKeys,
+    plugins: pluginsOption,
+    overwriteBaseKeys,
+  } = options
+
+  const baseKeyBinding = { ...baseKeymap }
+  overwriteBaseKeys?.forEach((key) => {
+    delete baseKeyBinding[key]
+  })
+
   const plugins = [
     buildInputRules(schema),
     keymap(buildKeymap(schema, mapKeys)),
-    keymap(baseKeymap),
+    keymap(baseKeyBinding),
```

Comment:
I just realized the keymap for Space and Enter would influence code editor as well if they're placed inside `buildKeymap`! So I remove them from `buildKeymap`, make them a new plugin and only use it in `createComponentEditorView`. But I have to add a new option `overwriteBaseKeys` here to prevent several keys from being handled by `baseKeymap` (i.e. Enter).

Reply:
Makes sense, although it feels like there should be an easier way! Maybe in the future we can copy in the `baseKeymap` definition from prosemirror and use it in the `buildKeymap`. That way overriding would work as expected.

Reply:
Yeah, I actually make a copy of `baseKeymap` for #173.
Is it okay if I include the fix for #173 in this PR?

Reply:
Sure! I have changes here but I think it will be easy to rebase.

Reply:
Got it! Do you wish to push your changes first? Or is it going to be easier for you if I push mine first 🤔?

Reply:
Hi @sampullman 

I've pushed the fix for #173 a couple minutes ago. There's a lint error related to the use of `os` in `base-keymap.ts`. I think `os` is only available in NodeJS, so I wonder if we could just comment out that part for now🤔?

Reply:
No problem, I saw that as well. I'll merge my changes and double check everything, then remove the `os` usage!